javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java

```diff
@@ -795,34 +793,99 @@ void apply(NodeText nodeText, Node node) {
+
+            int previousCsmElementIndex = csmElementListIterator.previousIndex();
+            CsmElement csmElement = csmElementListIterator.next();
+            int nextCsmElementIndex = csmElementListIterator.nextIndex();
+
+            Map<MatchClassification, Integer> potentialMatches = new EnumMap(MatchClassification.class);
+            for (int i = startIndex; i<nodeText.getElements().size(); i++){
+                if (!correspondingIndices.contains(i)) {
+                    TextElement textElement = nodeText.getTextElement(i);
+
+                    boolean isCorresponding = isCorrespondingElement(textElement, csmElement, node);
+
+                    if (isCorresponding) {
+                        boolean hasSamePreviousElement = false;
+                        if (i > 0 && previousCsmElementIndex > -1) {
+                            TextElement previousTextElement = nodeText.getTextElement(i - 1);
+
+                            hasSamePreviousElement = isCorrespondingElement(previousTextElement, elements.get(previousCsmElementIndex), node);
                         }
-                    }
-                } else if (csmElement instanceof CsmChild) {
-                    CsmChild csmChild = (CsmChild)csmElement;
-                    if (textElement instanceof ChildTextElement) {
-                        ChildTextElement childTextElement = (ChildTextElement)textElement;
-                        if (childTextElement.getChild() == csmChild.getChild()) {
-                            usedIndexes.add(i);
-                            return i;
+
+                        boolean hasSameNextElement = false;
+                        if (i < nodeText.getElements().size() - 1 && nextCsmElementIndex < elements.size()) {
+                            TextElement nextTextElement = nodeText.getTextElement(i + 1);
+
+                            hasSameNextElement = isCorrespondingElement(nextTextElement, elements.get(nextCsmElementIndex), node);
+                        }
+
+                        if (hasSamePreviousElement && hasSameNextElement) {
+                            potentialMatches.putIfAbsent(MatchClassification.ALL, i);
+                        } else if (hasSamePreviousElement) {
+                            potentialMatches.putIfAbsent(MatchClassification.PREVIOUS_ONLY, i);
+                        } else if (hasSameNextElement) {
+                            potentialMatches.putIfAbsent(MatchClassification.NEXT_ONLY, i);
+                        } else {
+                            potentialMatches.putIfAbsent(MatchClassification.SAME_ONLY, i);
                         }
                     }
-                } else {
-                    throw new UnsupportedOperationException();
                 }
             }
+
+            // Prioritize the matches from best to worst
+            Optional<MatchClassification> bestMatchKey = potentialMatches.keySet().stream()
```

Comment:
sorting the potential matches according to their priorization and keeping the best match.